### PR TITLE
PLANET-3320: [ENBLOCK] Add a new dataLayer field (Goal) on Form Block backend

### PR DESCRIPTION
**Pull request by pablocubico**
Adding a `gGoal` field to dataLayer for the new `enblock`.

Ref: https://jira.greenpeace.org/browse/PLANET-3320

### CODE REVIEW

File: classes/controller/blocks/class-enblock-controller.php [hunk 1]
```diff
@@ -175,6 +183,16 @@ function ( $a, $b ) {
 					],
 					'options'     => $pages_options,
 				],
+				[
+					'label'       => __( 'Goal', 'planet4-engagingnetworks' ),
+					'attr'        => 'goal',
```
Comment (angtheod) [inline classes/controller/blocks/class-enblock-controller.php]: Shouldn't this value be `enform_goal` ?

File: includes/enblock.twig [hunk 2]
```diff
@@ -27,6 +27,8 @@
 								<input type="hidden" name="en_page_id" value="{{ fields.en_page_id }}">
 								<input type="hidden" name="thankyou_title" value="{{ fields.thankyou_title }}">
 								<input type="hidden" name="thankyou_subtitle" value="{{ fields.thankyou_subtitle }}">
+								<input type="hidden" id="enform_goal" value="{{ fields.goal }}">
```
Comment (angtheod) [inline includes/enblock.twig]: Could you please add also the name attribute here? `name="enform_goal"`

File: assets/js/partials/enform_submit.js [hunk 3]
```diff
@@ -161,9 +161,15 @@ $(document).ready(function () {
 
           // DataLayer push event on EN form submission.
           if ( typeof google_tag_value !== 'undefined' && google_tag_value ) {
-            dataLayer.push({
+            var dataLayerPayload = {
               'event' : 'petitionSignup'
-            });
+            };
+
+            if ( $('#enform_goal').val() ) {
```
Comment (angtheod) [inline assets/js/partials/enform_submit.js]: Could store here the value and use it inside the if statement to avoid duplicate selector.

**Comment by angtheod** [comment]
@pablocubico there is a conflict in the PR

**Comment by pablocubico** [comment]
@angtheod oh right, I need to address these! will do in a bit

**Comment by pablocubico** [comment]
@angtheod Updated!

**Pull request merged**